crates/polars-core/src/chunked_array/ops/aggregate/mod.rs

```diff
@@ -475,6 +475,49 @@ impl ChunkAggSeries for StringChunked {
     }
 }
 
+#[cfg(feature = "dtype-categorical")]
+impl CategoricalChunked {
+    fn min_categorical(&self) -> Option<&str> {
+        if self.is_empty() || self.null_count() == self.len() {
+            return None;
+        }
+        if self.uses_lexical_ordering() {
+            self.iter_str()
+                .flatten()
+                .reduce(|acc, val| if val < acc { val } else { acc })
+        } else {
+            let ca_phys = self.physical();
+            let min_idx = ca_phys.get(ca_phys.first_non_null().unwrap())?;
+            self.get_rev_map().get_categories().get(min_idx as usize)
```

Comment:
Careful here--with enums, the rev map can be bigger than the unique items in the Series, if I'm not mistaken:

```python
s = pl.Series(["c", "d"], dtype=pl.Enum(["a", "b", "c", "d"]))
```
The min category is "a" but the min value in the series is "c".

I'm not sure if the revmap gets revised for slices, so this may also be the case with normal categoricals:

```python
# revmap will be a, b, c, d
s = pl.Series(["a", "b", "c", "d"], dtype=pl.Categorical)

# s2 is now c, d. Does the rev map still have a & b?
s2 = s.slice(2)
```

Reply:
In the first example, `rev_map` can be bigger but that shouldn't matter. We find the index of the first (last for `max`) non-null value in the series and then get the string that corresponds to that index. And this is actually the intented behaviour: `min` returns the minimum value of the series, regardless of what the rev_map contains.

In the second example, it seems that it does get updated:
```python
s = pl.Series(["a", "b", "c", "d"], dtype=pl.Categorical)
s2 = s.slice(2)

s2
shape: (2,)
Series: '' [cat]
[
        "c"
        "d"
]

s2.min()
# "c"

s2.max()
# "d"
```

Reply:
Ahh indeed. I misread the `self.get_rev_map().get_categories().get(min_idx as usize)` as "get the minimum category. Carry on!

Reply:
Thanks for the review and the comments!